x-pack/plugins/actions/server/actions_client.ts

```diff
@@ -139,21 +171,32 @@ export class ActionsClient {
    * Update action
    */
   public async update({ id, action }: UpdateOptions): Promise<ActionResult> {
-    await this.authorization.ensureAuthorized('update');
-
-    if (
-      this.preconfiguredActions.find((preconfiguredAction) => preconfiguredAction.id === id) !==
-      undefined
-    ) {
-      throw new PreconfiguredActionDisabledModificationError(
-        i18n.translate('xpack.actions.serverSideErrors.predefinedActionUpdateDisabled', {
-          defaultMessage: 'Preconfigured action {id} is not allowed to update.',
-          values: {
-            id,
-          },
-        }),
-        'update'
+    try {
+      await this.authorization.ensureAuthorized('update');
+
+      if (
+        this.preconfiguredActions.find((preconfiguredAction) => preconfiguredAction.id === id) !==
+        undefined
+      ) {
+        throw new PreconfiguredActionDisabledModificationError(
```

Comment:
Is it intentional to capture the errors thrown when the action type is disabled? If so, there's a few other places (`this.actionTypeRegistry.ensureActionTypeEnabled`) that will throw similar errors.

Reply:
When does this error occur? Is it role based or just some kind of global configuration? 

I thought it was an authorisation error but looks like I made a mistake here.

Reply:
> When does this error occur? Is it role based or just some kind of global configuration?

There are two types of actions in the system, pre-configured and dynamic / user created actions. This code will throw whenever the update API is called for a pre-configured action. Since those actions are configured via kibana.yml, we don't support updates and throw the error.

So I think we may be ok here to skip auditing this error?

Reply:
I've double checked with Oleg and we think this is audit worthy. As an auditor you'd want to know why someone is trying to update preconfigured connectors. 

I'll make `ensureActionTypeEnabled` error are included in the audit log. 

Are there any other errors I should include?

Reply:
Actually, looking at the errors thrown inside `ensureActionTypeEnabled`, I'm not sure they're audit worthy. 

The first type of error is thrown when an action type is disabled. That just feels like a validation error. 

The second type of error is thrown when the license doesn't allow a certain type. That's definitely not security related /  audit worthy. 

I'd say let's leave these as is until we get a requirement to include those errors.

Reply:
👍 works for me!